web/src/layouts/MainLayout/MainLayout.js

```diff
@@ -85,6 +87,34 @@ const MainLayout = ({ children }) => {
       previousUserID = currentUser
     }
   }, [data, currentUser, isAuthenticated])
+  const hash = window.location.hash
+  useEffect(() => {
```

Comment:
This effect is big and unruly enough that I probably should have pulled it out into its own custom hook. The use effect has been added to the nav bar since it's on every page, but in reality, I should only ever be needed on the home page since that's where the verification emails redirect to.

Essentially what it's doing is looking for a hash param that's either a confirmation_token or a recovery_token.

If it's a confirmation token it simply sends that off to gotrue, nothing more to do.

If it's a recover token, it sends to token to gotrue and from there re-directs them to the update password page.